vm/src/stdlib/ctypes/array.rs

```diff
@@ -20,3 +79,141 @@ impl PyValue for PyCArray {
+                    } else {
+                        Ok(usize::try_from_object(vm, length_obj)
+                            .or(Err(vm.new_overflow_error(
+                                "The '_length_' attribute is too large".to_string(),
+                            )))?)
+                    }
+                } else {
+                    Err(vm
+                        .new_type_error("The '_length_' attribute must be an integer".to_string()))
+                }
+            }
+            Err(_) => {
+                Err(vm.new_attribute_error("class must define a '_type_' _length_".to_string()))
+            }
+        }?;
+
+        if let Ok(outer_type) = vm.get_attribute(cls.as_object().to_owned(), "_type_") {
+            match vm.get_attribute(outer_type, "_type_") {
+                Ok(inner_type)
+                    if vm.issubclass(&inner_type.clone_class(), &PyCPointer::static_type())?
+                        || vm.issubclass(
+                            &inner_type.clone_class(),
+                            &PySimpleType::static_type(),
+                        )? =>
+                {
+                    let subletter = vm
+                        .get_attribute(outer_type, "_type_")?
+                        .downcast_exact::<PyStr>(vm)
+                        .unwrap()
+                        .to_string();
+
+                    let itemsize = get_size(subletter.as_str());
+
+                    let myself = PyCArray {
+                        _type_: subletter,
+                        _length_: length,
+                    }
+                    .into_ref_with_type(vm, cls)?;
+
+                    vm.set_attr(
+                        myself.as_object(),
+                        "_buffer",
+                        PyRwLock::new(RawBuffer {
+                            inner: Vec::with_capacity(length * itemsize).as_mut_ptr(),
+                            size: length * itemsize,
+                        }),
+                    )?;
+
+                    Ok(myself)
+                }
+                _ => Err(vm.new_type_error("_type_ must have storage info".to_string())),
+            }
+        } else {
+            Err(vm.new_attribute_error("class must define a '_type_' attribute".to_string()))
+        }
+    }
+
+    #[pymethod(magic)]
+    pub fn init(&self, value: FuncArgs, vm: &VirtualMachine) -> PyResult<()> {
+        // @TODO
+        Ok(())
+    }
+
+    #[pyproperty(name = "value")]
+    pub fn value(&self, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        let obj = self.into_object(vm);
+        let buffer = try_buffer_from_object(vm, &obj)?;
+
+        let res = if self._type_ == "u" {
+            vm.new_pyobj(
+                if cfg!(windows) {
+                    WideString::from_vec(
+                        buffer
+                            .obj_bytes()
+                            .chunks(4)
+                            .map(|c| NativeEndian::read_u32(c))
```

Comment:
You can use `u32::from_ne_bytes` for this; also, there's already a byteorder dependency in vm/Cargo.toml

Reply:
Also, what function is this supposed to be? CArray doesn't have a value property from what I can see

Reply:
> You can use `u32::from_ne_bytes` for this; also, there's already a byteorder dependency in vm/Cargo.toml

Oh boy, I missed that one, I've found the `bt`and `le` counterparts. I can simply ignore the whole `ByteOrder` crate, then.

> Also, what function is this supposed to be? CArray doesn't have a value property from what I can see

These properties come from it's metaclass (`PyCArrayType_Type`),  which are `CharArray_getsets` and `WCharArray_getsets`